### DISCUSSION TITLE
Extensible records

### DISCUSSION THREAD

**Discussion by leolimasa** [koka-lang/koka#454] [General]
The koka book has a section on "Extensible Data Types" which is just marked as a TODO.

Does that mean koka supports extensible types? If it does, what does that entail - is it like elm's extensible records? Are there any samples?

I've just stumbled into koka and I'm seriously loving it. It is a really well thought out language.

**Comment by TimWhiting** [comment]
Here is a good example and use case of extensible data types (definitely different than extensible records). https://github.com/koka-lang/koka/discussions/452#discussioncomment-8275272

I'm glad you are enjoying Koka!

It looks like Elm is using Daan's paper on [Extensible records with scoped labels](https://www.microsoft.com/en-us/research/publication/extensible-records-with-scoped-labels/): https://elm-lang.org/docs/advanced-topics

As far as I understand it that is what is called extensible structural types. In Koka there is only nominal types currently, and so extensible data types refer to types that we can add new named constructors(variants) for. Every type has at least one named constructor, and anonymous structs are not available. I'd be curious to know if Daan would consider adding anonymous extensible structs. @daanx

**Comment by leolimasa** [comment]
That's very helpful. It makes sense now. Thank you for the speedy response!
